pkg/processor/timeout/timeout.go

```diff
@@ -81,26 +79,14 @@ func (w EventTimeoutWatcher) watch() {
 					workerInstance := workerInstance
 
 					workerErrGroup.Go("Watch Event Timeout", func() error {
-						eventTime := workerInstance.GetEventTime()
-						if eventTime == nil {
-							return nil
-						}
-
-						elapsedTime := now.Sub(*eventTime)
-						if elapsedTime <= w.timeout {
+						restartRequired := workerInstance.RestartRequired()
+						if !restartRequired {
```

Comment:
There is a possible delay between the event timing out and this goroutine restarting the worker, because now there are multiple places that can set the `RestartRequired` status, but this check is only happening every certain interval.
What happens in this delay? what if a new event arrives?

Reply:
@TomerShor if new event arrives, it is processed by another connection

Reply:
But when a worker is restarted, does it only restart the connections that require it?